packages/notebook/src/model.ts

```diff
@@ -116,6 +114,8 @@ export class NotebookModel implements INotebookModel {
 
     (this.sharedModel as models.YNotebook).dirty = false;
     this.sharedModel.changed.connect(this._onStateChanged, this);
+    (this.sharedModel as models.YNotebook).nbformat = this._nbformat;
```

Comment:
I feel that `nbformat` should be initialized in the `shared-model` package. If we initialize it here, we will set `nbformat` every time this class is instantiated (i.e. every time a new user joins).

The same is true for the `dirty` flag above.

Here, we should instead only check whether `this.nbformat === sharedModel.nbformat` and throw an error if this is not true.

Reply:
I can open another PR to initialize everything in the shared model

Reply:
This is kind of a big issue. `nbformat` must only be initialized in the `create` and `createStandalone` methods. Can you please revert the changes in this PR? 

I added some comments about why we should avoid casting to the concrete implementation whenever possible:
https://github.com/jupyterlab/jupyterlab/pull/11525/files#r771420793

Please also check other places where you casted to the concrete Yjs implementation.